app/src/utils/data/ipfs.ts

```diff
@@ -14,7 +14,8 @@ function assertIPFSPointer(logoPtr: MetaPtr | undefined) {
     logoPtr = decodeMetadataId(logoPtr);
   }
   const protocol = BigNumber.from(logoPtr.protocol).toString();
-  if (protocol !== '1') throw new Error(`assertIPFSPointer: Expected protocol ID of 1, found ${protocol}`);
+  if (protocol !== '1' && protocol !== '0' && !logoPtr.pointer)
```

Comment:
Hmm, so as written, if anyone were to upload/update a grant with a metaPtr that had a value other than `1` or `0`, or with an empty pointer, this method would throw. Since we call this method in `getMetaPtr`, and use that method to fetch metadata throughout the app, then if anyone were to do this, the app would start throwing errors.

I don't believe an unsupported `protocol` value, or empty/invalid `metaPtr` value, should not result in the app throwing errors at any point, since users could put anything in there, and we can't control that. I think it should always degrade gracefully in some way, for example by simply excluding the grant if content fails to load. This might require a different PR though. Curious to hear what you think @mds1.

Reply:
Generally agreed! Though this file isn't the right place to handle that graceful degradation based on how I was thinking of the architecture.

For context: `getMetaPtr` should primarily only be called by the `ptrToURI` wrapper method, with an exception for new grant / image upload code paths that call it directly because they're currently tightly coupled to IPFS. My intent with this `assertIPFSPointer` was to make sure only IPFS pointers are being passed into the methods in this file, and this file should be limited to handling cases where `protocol` is 1. Therefore if we throw in this method that's a bug and in that case I think it's ok to fail loudly. Given that, this line should only be `if (protocol !== '1') throw new Error(...)` and this entire file should not care about `protocol` being any non-1 value

Given all that, and if you agree with that architecture decision of isolating this file to strictly deal with IPFS, then `ptrToURI` and other methods in `utils.ts` are where we should gracefully handle the failure.